dialog/input.go

```diff
@@ -0,0 +1,81 @@
+package dialog
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// InputDialog is a variation of a dialog which prompts the user to enter some
+// text, with an optional validation function.
+type InputDialog struct {
+	*dialog
+
+	onConfirm func(string)
+
+	entry *widget.Entry
+}
+
+// SetText changes the current text value of the input dialog, this can
+// be useful for setting a default value.
+func (i *InputDialog) SetText(s string) {
+	i.entry.SetText(s)
+}
+
+// NewInput creates a dialog over the specified window for the user to enter
+// a value, possibly with validation.
+//
+// onConfirm is a callback that runs when the user enters a valid string of
+// text and clicks the "confirm" button. May be nil.
+func NewInput(title, message string, onConfirm func(string), parent fyne.Window) *InputDialog {
+
+	// create the widgets necessary for the dialog
+	entry := widget.NewEntry()
+
+	// content container for our widgets
+	content := widget.NewHBox(
+		widget.NewLabel(message),
+		entry,
+	)
+
+	// instantiate the dialog, and override the content
+	d := newDialog(title, message, theme.QuestionIcon(), func(response bool) {}, parent)
+	d.content = content
+
+	// hide the dialog, and empty the text
+	d.dismiss = widget.NewButton("Cancel", func() {
+		entry.Text = ""
+		d.Hide()
+	})
+	d.dismiss.Icon = theme.CancelIcon()
+
+	// we will override the OnTapped function later once we've instanced
```

Comment:
I don't see why this is needed - if you define the inputdialog first then all you reference is the entry, which is already defined.

Reply:
I think my thought at the time was that `onConfirm` may have been changed after the fact, but that cannot happen since it is a private field.

Fixed in 87558a48